oop/src/components/DateInput.vue

```diff
@@ -77,6 +77,13 @@ export const afterDateValidator = (compareDateName) => {
   };
 };
 
+export const sameDateValidator = (compareDateName) => {
+  return (date, vm) => {
+    const dateToCompare = vm[compareDateName];
+    return (!isBefore(date, dateToCompare) && !isAfter(date, dateToCompare));
```

Comment:
Instead of using `isBefore` and `isAfter`, you can use `isSameDate` from the same package.

https://date-fns.org/v2.18.0/docs/isSameDay